app/models/setting/general.rb

```diff
@@ -1,15 +1,22 @@
 class Setting::General < Setting
   include UrlValidator
 
+  # Lazy-load this to avoid loading this during Rails startup
+  def self.locales
+    Hash['' => _("Browser locale")].merge(Hash[FastGettext.human_available_locales.map { |lang| [lang[1], lang[0]] }])
```

Comment:
Doesn't this mean it gets calculated every time the setting is used? Isn't there some way to cache the result or does rails already do so?

Reply:
Well `FastGettext.human_available_locales` is cached (it's a global), this is just a helper which constructs a hash when Setting page is loaded. Nothing we should care optimizing I think.